napari/components/add_layers_mixin.py

```diff
@@ -602,11 +606,13 @@ def add_shapes(
             The newly-created shapes layer.
         """
         if data is None:
-            ndim = max(self.dims.ndim, 2)
+            if ndim is None:
```

Comment:
This default was added to make https://github.com/napari/napari/blob/master/napari/_qt/qt_viewer_buttons.py#L43 clean, but we could now make that pass `ndim=viewer.dims.ndim` and then rely on the default behavior inside the layer itself. I think I might prefer that. Thoughts @jni?

Reply:
I suggest making this a follow-up PR

Reply:
That's fine with me! I will approve this now then :-)